src/number/complete.rs

```diff
@@ -1546,30 +1620,62 @@ where
   T: AsBytes,
   T: for<'a> Compare<&'a [u8]>,
 {
-  let (i, (sign, integer, fraction, exponent)) = recognize_float_parts(input)?;
+  let (input, sign) = sign(input.clone())?;
+  let res = float_finite::<T, E>(input.clone());
+  let (i, mut float) = res.or(float_nonfinite::<T, E>(input))?;
```

Comment:
It will always call `float_nonfinite`; "Arguments passed to or are eagerly evaluated; if you are passing the result of a function call, it is recommended to use or_else, which is lazily evaluated." https://doc.rust-lang.org/std/result/enum.Result.html#method.or

Reply:
Ok fixed as of the latest commit.